chainer/functions/normalization/decorrelated_batch_normalization.py

```diff
@@ -0,0 +1,253 @@
+from chainer.backends import cuda
+from chainer import function
+from chainer import function_node
+from chainer.utils import argument
+from chainer.utils import type_check
+
+
+class DecorrelatedBatchNormalizationFunction(function_node.FunctionNode):
+
+    def __init__(self, groups=16, eps=2e-5, mean=None, projection=None,
+                 decay=0.9):
+        self.groups = groups
+
+        self.expected_mean = mean
+        self.expected_projection = projection
+
+        self.eps = eps
+        self.decay = decay
+        self.axis = None
+
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 1)
+        x_type = in_types[0]
+        type_check.expect(
+            x_type.dtype.kind == 'f',
+            x_type.shape[1] % self.groups == 0,
+        )
+        type_check.expect(
+            x_type.ndim >= 2,
+        )
+
+    def forward(self, inputs):
+        self.retain_inputs(())
+        x = inputs[0]
+        xp = cuda.get_array_module(x)
+        spatial_ndim = len(x.shape[2:])
+        spatial_axis = tuple(range(2, 2 + spatial_ndim))
+        b, c = x.shape[:2]
+        g = self.groups
+        C = c // g
+        m = b * g
+        for i in spatial_axis:
+            m *= x.shape[i]
+
+        if self.expected_mean is None:
+            self.expected_mean = xp.zeros(C, dtype=x.dtype)
+            self.expected_projection = xp.eye(C, dtype=x.dtype)
+
+        if self.groups > 1:
+            x = x.reshape((b * g, C, ) + x.shape[2:])
+        x_hat = x.transpose((1, 0) + spatial_axis).reshape((C, -1))
+
+        mean = x_hat.mean(axis=1)
+        x_hat = x_hat - mean[:, None]
+        self.eps = x.dtype.type(self.eps)
+
+        I = self.eps * xp.eye(C, dtype=x.dtype)
+        cov = x_hat.dot(x_hat.T) / x.dtype.type(m) + I
+        self.eigvals, self.eigvectors = xp.linalg.eigh(cov)
+        U = xp.diag(self.eigvals ** -0.5).dot(self.eigvectors.T)
+        self.y_hat_pca = U.dot(x_hat)  # PCA whitening
+        y_hat = self.eigvectors.dot(self.y_hat_pca)  # ZCA whitening
+
+        y = y_hat.reshape((C, b * g,) + x.shape[2:]).transpose(
+            (1, 0) + spatial_axis)
+        if self.groups > 1:
+            y = y.reshape((-1, c, ) + x.shape[2:])
+
+        # Update running statistics
+        q = x.size // C
+        adjust = q / max(q - 1., 1.)  # unbiased estimation
+        self.expected_mean *= self.decay
+        self.expected_mean += (1 - self.decay) * mean
+        self.expected_projection *= self.decay
+        projection = self.eigvectors.dot(U)
+        self.expected_projection += (1 - self.decay) * adjust * projection
+
+        return y,
+
+    def backward(self, indexes, grad_outputs):
+        gy, = grad_outputs
+
+        f = DecorrelatedBatchNormalizationGrad(
+            self.groups, self.eigvals, self.eigvectors, self.y_hat_pca)
+        return f(gy),
+
+
+class DecorrelatedBatchNormalizationGrad(function.Function):
```

Comment:
Let's use the new `FunctionNode` (only need to change the above `f(gy),` to `f.apply((gy,))`.